### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by meis0n**

### CODE REVIEW

**Comment by AinuraAliieva** [comment]
@meis0n Please merge master into the current branch

File: src/app/shared/components/header/header.component.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <div class="app-header">
     <div class="app-header-container container">
         <app-logo class="mr-auto"></app-logo>
-        <div class="button button--transparent"><span class="icon icon-user"></span>{{login}}</div>
-        <div class="button button--transparent"><span class="icon icon-exit"></span>Log off</div>
+        <div *ngIf="isAuthenticated" class="button button--transparent" (click)="onUserInfo($event)"><span class="icon icon-user"></span>{{login}}</div>
+        <div *ngIf="isAuthenticated" class="button button--transparent" (click)="onLogoff($event)"><span class="icon icon-exit"></span>Log off</div>
```
Comment (AinuraAliieva) [inline src/app/shared/components/header/header.component.html]: I suggest optimizing this part by using ng-container with one subscription.
In that case, we reduce subscriptions from two to one :)

something like that

```
<ng-container *ngIf="isAuthenticated">
  <div> ...
  <div> ...
```

File: src/app/shared/services/authorization.service.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+export class AuthorizationService {
+  login(user: User): void {
+    this.localStorage.set(USER_LOGIN_DATA_KEY, user);
+  }
+  logout(): void {
+    this.localStorage.reset(USER_LOGIN_DATA_KEY);
+  }
+  isAuthenticated(): boolean {
+    const user = this.localStorage.get(USER_LOGIN_DATA_KEY);
+    return Boolean(user);
```
Comment (AinuraAliieva) [inline src/app/shared/services/authorization.service.ts]: we could use `getCurrentUserInfo` here and follow DRY

File: src/app/shared/components/header/header.component.html (hunk 1 above)
Comment (meis0n) [reply src/app/shared/components/header/header.component.html]: done

File: src/app/shared/services/authorization.service.ts (hunk 2 above)
Comment (meis0n) [reply src/app/shared/services/authorization.service.ts]: done

**Pull request merged**